### PR TITLE
[Tansformers] Export script refactor

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR represents the main feature branch for the upcoming export script refactors.

The high level goal for this push are:
- To allow users to export sample/inputs outputs using `sparseml.transformers.export_onnx` utility
- To use the same dataloaders as training script for exporting sample inputs/outputs

To this end the PRs have been broken out in the following manner to allow easier reviews

- [x] Changes needed for in `src/sparseml/transformers/export.py` to allow dataset creation and using the `Trainer'(s)` `save_sample_inputs_and_outputs` function (CURRENT PR)
    - [x] Adding additional arguments: namely `num_export_samples`  and a `dataset_name`
    - [x] Adding a stub `load_task_dataset` function to get dataset based on task, tokenizer and data_args
    - [x] Tweaking `Trainer` instance creation by adding an `eval_dataset` this will be used for fetching dataloader

- [x] Changes needed to support `masked_language_modeling` ( PR: #1116 )
    - [x] Refactor existing script, create importable function for fetching the dataset
    - [x] Requisit changes in the `load_task_dataset` function
    
- [x] Changes needed to support `question_answering` (PR: #1135 )
    - [x] Refactor existing script, create importable function for fetching the dataset
    - [x] Requisit changes in the `load_task_dataset` function

- [x] Changes needed to support `token_classification` (PR: #1132 )
    - [x] Refactor existing script, create importable function for fetching the dataset
    - [x] Requisit changes in the `load_task_dataset` function

- [x] Changes needed to support `text_classification` (PR: #1137 )
    - [x] Refactor existing script, create importable function for fetching the dataset
    - [x] Requisit changes in the `load_task_dataset` function
    
Note: All PRs must be first merged into this feature branch and the tests should be run, after which this feature branch can be landed on main 

Tests:
- [X] QA
```bash
sparseml.transformers.export_onnx --task qa --model_path '/home/XXXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/training' \
--num_export_samples 10 \
--data_args '{"dataset_name": "squad"}'
```

Output:
```bash
2022-11-16 14:03:41 sparseml.transformers.sparsification.trainer INFO     Exporting 10 samples to tmp_trainer
2022-11-16 14:03:45 sparseml.transformers.sparsification.trainer INFO     Exported 10 samples to tmp_trainer
2022-11-16 14:03:45 sparseml.transformers.export INFO     10 sample inputs/outputs exported
2022-11-16 14:03:45 sparseml.transformers.export INFO     Saved model.onnx in the deployment folder at /home/XXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/deployment/model.onnx
2022-11-16 14:03:45 sparseml.transformers.export INFO     Saved tokenizer.json in the deployment folder at /home/XXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/deployment/tokenizer.json
2022-11-16 14:03:45 sparseml.transformers.export INFO     Saved tokenizer_config.json in the deployment folder at /home/XXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/deployment/tokenizer_config.json
2022-11-16 14:03:45 sparseml.transformers.export INFO     Saved config.json in the deployment folder at /home/XXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/deployment/config.json
2022-11-16 14:03:45 sparseml.transformers.export INFO     Created deployment folder at /home/XXX/.cache/sparsezoo/15a1359c-6389-4ffd-ab06-d44a945159c2/deployment with files: ['tokenizer.json', 'model.onnx', 'tokenizer_config.json', 'config.json']
```

```bash
tmp_trainer
├── runs
│   └── Nov16_14-03-15_mark-SYS-5039AD-I
│       └── events.out.tfevents.1668625397.XXXXXXXXXXX
├── sample_inputs
│   ├── inp-0000.npz
│   ├── inp-0001.npz
│   ├── inp-0002.npz
│   ├── inp-0003.npz
│   ├── inp-0004.npz
│   ├── inp-0005.npz
│   ├── inp-0006.npz
│   ├── inp-0007.npz
│   ├── inp-0008.npz
│   └── inp-0009.npz
└── sample_outputs
    ├── out-0000.npz
    ├── out-0001.npz
    ├── out-0002.npz
    ├── out-0003.npz
    ├── out-0004.npz
    ├── out-0005.npz
    ├── out-0006.npz
    ├── out-0007.npz
    ├── out-0008.npz
    └── out-0009.npz

```

### CODE REVIEW

**Comment by dbogunowicz** [comment]
So far so good!

File: src/sparseml/transformers/export.py [hunk 1]
```diff
@@ -170,11 +200,17 @@ def export_transformer_to_onnx(
     )
     model = load_task_model(task, model_path, config)
     _LOGGER.info(f"loaded model, config, and tokenizer from {model_path}")
-
-    model = model.train()
```
Comment (corey-nm) [inline src/sparseml/transformers/export.py:174]: I think we still need this (See #1085)

File: src/sparseml/transformers/export.py [hunk 2]
```diff
@@ -126,13 +131,29 @@ def load_task_model(task: str, model_path: str, config: Any) -> Module:
...
+
 def export_transformer_to_onnx(
     task: str,
     model_path: str,
     sequence_length: int = 384,
     convert_qat: bool = True,
     finetuning_task: Optional[str] = None,
     onnx_file_name: str = MODEL_ONNX_NAME,
+    num_export_samples: int = 0,
+    dataset_name: Optional[str] = None,
```
Comment (bfineran) [inline src/sparseml/transformers/export.py:156]: need to consider making this a dict of `DataArgs` as there may be more script specific arguments to load datasets (especially around custom datasets). can import the data arg classes by script specifically for parsing

File: src/sparseml/transformers/export.py (hunk 1 above)
Comment (rahul-tuli) [reply src/sparseml/transformers/export.py:174]: Added back in, great catch!

**Review by bfineran — APPROVED** [review]

**Review by corey-nm — APPROVED** [review]
lgtm, nice diff!

**Pull request merged**